### DISCUSSION TITLE
SImple proxying

### DISCUSSION THREAD

**Discussion by florian-lefebvre** [fingerprintjs/fingerprintjs#961] [Q&A] [answered]
Looking at the docs, there are several guides to enable proxying for specific solutions (Azure, aws etc).

However, I'd like to setup something a bit easier with Netlify redirects or even a custom Astro endpoint. Closest thing I found is https://dev.fingerprint.com/docs/custom-subdomain-setup but I'd ideally want it at `domain.com/api/fp`. Is that possible? Other services like Sentry, mixpanel etc allow such usage.

**Comment by JuroUhlar** [comment, marked as answer]
Hi @florian-lefebvre! 

So nothing is stopping you from implementing a custom proxy solution, all our integrations are open-source, so you can use them as a base, for example, here is the Cloudflare integration worker: https://github.com/fingerprintjs/fingerprintjs-pro-cloudflare-worker

But I would advise against it. Proxying requests to Fingerprint APIs is a little more complicated compared to services like Sentry or Mixpanel: 
* You need to proxy multiple types of requests (agent download, identification + other requests used to gather additional entropy)
* You need to correctly handle caching, IP address forwarding, and setting request headers
* The data contract between the JS agent and our APIs can change frequently as we improve our identification methods and keep up with browser updates. These changes can require updating the proxy functions. The data contract is not stable, nor public so you might not even know that something has changed and how to conform. 

Getting the proxy implementation wrong or just not keeping up with our updates can disrupt visitor identification completely or just silently lead to worse accuracy. We have limited resources to support and troubleshoot custom proxy implementations. I would recommend using the [Custom subdomain setup](https://dev.fingerprint.com/docs/custom-subdomain-setup) or one of our cloud proxy integrations, if you can. Most are easier to install than coding a custom proxy endpoint (especially Cloudflare and custom subdomain setup) and we handle all the updates for you.

If you do require a custom solution, please reach out to our [support team](https://fingerprint.com/support/) who can point you in the right direction given your exact use case and tech stack.

**Reply by florian-lefebvre** [reply]
Alright I'll go for the subdomain then, thx for the quick answer!
